.github/workflows/update-prerun-script.yml

```diff
@@ -0,0 +1,86 @@
+name: Update pre build script on Saucelabs
+
+on:
+  schedule:
+    - cron: '*/10 * * * *'
+#  schedule:
+#    - cron: '0 0 1 */2 *'
+permissions:
+  contents: read
+  pull-requests: write
+
+env:
+  SAUCE_USERNAME: ${{secrets.SAUCE_USERNAME}}
+  SAUCE_ACCESS_KEY: ${{secrets.SAUCE_ACCESS_KEY}}
+
+jobs:
+  update-pre-run-script:
+    name: Update pre run script on Saucelabs
+    runs-on: ubuntu-latest
+    outputs:
+      script-id: ${{ steps.upload-pre-run-script.outputs.script-id }}
+    steps:
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+      - name: Copy script from S3 bucket
+        run: aws s3 cp s3://saucelabs-prebuild-script/script.sh ./
+      - name: Upload pre run script to Saucelabs
+        id: upload-pre-run-script
+        run: |
+          RESPONSE="$(curl -u "$SAUCE_USERNAME:$SAUCE_ACCESS_KEY" —location --request POST 'https://api.us-west-1.saucelabs.com/v1/storage/upload' --form 'payload=@"script.sh"' --form 'name="blur_replacement_script"')"
+          SCRIPT_ID=$(jq -r '.item.id' <<< "$RESPONSE")
+          rm -rf script.sh
+          echo "::set-output name=script-id::$SCRIPT_ID"
+      
+  create-pr:
+    name: Create PR for origin branch
+    runs-on: ubuntu-latest
```

Comment:
If this job depends on the previous job, doesnt it have to have the `needs: update-pre-run-script` property set?

Reply:
Added